File: _c++/symulacja_organizmow/Organizmy/Zwierzeta/Lis.cpp
//
//  Lis.cpp
//  WirtualnySwiat
//
//  Created by Michal Tarnacki on 10/04/2022.
//

#include "Lis.hpp"
Lis::Lis(Swiat &swiat):Zwierze(swiat){
    
}
void Lis::Akcja(){
    Zwierze::Akcja();
    for (Organizm *organizm: swiat.GetOrganizmy()){
        if(organizm->GetSila()>this->GetSila()&&organizm->GetPolozenie()==this->GetPolozenie()){
            this->SetPolozenie(this->GetPoprzedniePolozenie());
            std::cout << "Lis chcial isc na pozycje: " << organizm->GetPolozenie() << " ale wytropil: " << organizm->Rysowanie() << " i nie idzie dalej " << std::endl;
        }
    }
}
char Lis::Rysowanie() const{
    return 'L';
}
void Lis::SetInicjatywa(){
    this->inicjatywa=7;
}
void Lis::SetSila(){
    this->sila=3;
}
Organizm* Lis::Urodz(){
    return new Lis(this->swiat);
}
bool Lis::MozeBycPartnerem(Organizm& organizm){
    if(dynamic_cast<Lis*>(&organizm))
       return true;
    else
       return false;
}

File: _java/symulacja_organizmow/src/Symulacja/Zwierzeta/Lis.java
package Symulacja.Zwierzeta;
import Symulacja.*;

import java.awt.*;

public class Lis extends Zwierze{
    protected void SetInicjatywa(){
        this.inicjatywa=7;
    };
    protected void SetSila(){
        this.sila=3;
    };
    public Lis(Swiat swiat){
        super(swiat);
        SetInicjatywa();
        SetSila();
    };
    public char Rysowanie(){
        return 'L';
    };
    public Organizm Urodz() {
        return new Lis(this.swiat);
    };
    public boolean MozeBycPartnerem(Organizm organizm){
        if(organizm instanceof Lis)
        return true;
    else
        return false;
    };
    public void Akcja(){
        super.Akcja();
        for (Organizm organizm: swiat.GetOrganizmy()){
            if(organizm.GetSila()>this.GetSila()&&organizm.GetPolozenie().equals(this.GetPolozenie())){
                this.SetPolozenie(this.GetPoprzedniePolozenie());
                swiat.zdarzenia.DodajString( "Lis chcial isc na pozycje: " + organizm.GetPolozenie() + " ale wytropil: " + organizm.Rysowanie() + " i nie idzie dalej ");
            }
        }

    }
    @Override
    public Color GetColor(){
        return new Color(0,100,0);
    };
}
